### DISCUSSION TITLE
Install Woes

### DISCUSSION THREAD

**Discussion by davegd1387** [alias-rahil/lyrics-searcher#12] [General]
I hope someone can help - I'm a newbie and am getting the following errors trying to install.
Thank you!

**npm ERR! code 1
npm ERR! path /home/dgd/nodejs/lib/node_modules/lyrics-searcher/node_modules/iconv
npm ERR! command failed
npm ERR! command sh -c node-gyp rebuild
npm ERR! make: Entering directory '/home/dgd/nodejs/lib/node_modules/lyrics-searcher/node_modules/iconv/build'
npm ERR!   CC(target) Release/obj.target/iconv/binding.o
npm ERR! make: Leaving directory '/home/dgd/nodejs/lib/node_modules/lyrics-searcher/node_modules/iconv/build'
npm ERR! gyp info it worked if it ends with ok
npm ERR! gyp info using node-gyp@8.4.1
npm ERR! gyp info using node@16.14.2 | linux | x64
npm ERR! gyp info find Python using Python version 3.8.10 found at "/usr/bin/python3"
npm ERR! gyp info spawn /usr/bin/python3
npm ERR! gyp info spawn args [
npm ERR! gyp info spawn args   '/home/dgd/nodejs/lib/node_modules/npm/node_modules/node-gyp/gyp/gyp_main.py',
npm ERR! gyp info spawn args   'binding.gyp',
npm ERR! gyp info spawn args   '-f',
npm ERR! gyp info spawn args   'make',
npm ERR! gyp info spawn args   '-I',
npm ERR! gyp info spawn args   '/home/dgd/nodejs/lib/node_modules/lyrics-searcher/node_modules/iconv/build/config.gypi',
npm ERR! gyp info spawn args   '-I',
npm ERR! gyp info spawn args   '/home/dgd/nodejs/lib/node_modules/npm/node_modules/node-gyp/addon.gypi',
npm ERR! gyp info spawn args   '-I',
npm ERR! gyp info spawn args   '/home/dgd/.cache/node-gyp/16.14.2/include/node/common.gypi',
npm ERR! gyp info spawn args   '-Dlibrary=shared_library',
npm ERR! gyp info spawn args   '-Dvisibility=default',
npm ERR! gyp info spawn args   '-Dnode_root_dir=/home/dgd/.cache/node-gyp/16.14.2',
npm ERR! gyp info spawn args   '-Dnode_gyp_dir=/home/dgd/nodejs/lib/node_modules/npm/node_modules/node-gyp',
npm ERR! gyp info spawn args   '-Dnode_lib_file=/home/dgd/.cache/node-gyp/16.14.2/<(target_arch)/node.lib',
npm ERR! gyp info spawn args   '-Dmodule_root_dir=/home/dgd/nodejs/lib/node_modules/lyrics-searcher/node_modules/iconv',
npm ERR! gyp info spawn args   '-Dnode_engine=v8',
npm ERR! gyp info spawn args   '--depth=.',
npm ERR! gyp info spawn args   '--no-parallel',
npm ERR! gyp info spawn args   '--generator-output',
npm ERR! gyp info spawn args   'build',
npm ERR! gyp info spawn args   '-Goutput_dir=.'
npm ERR! gyp info spawn args ]
npm ERR! gyp info spawn make
npm ERR! gyp info spawn args [ 'BUILDTYPE=Release', '-C', 'build' ]
npm ERR! In file included from ../binding.c:17:
npm ERR! ../support/iconv.h:54:10: fatal error: errno.h: No such file or directory
npm ERR!    54 | #include <errno.h>
npm ERR!       |          ^~~~~~~~~
npm ERR! compilation terminated.
npm ERR! make: *** [iconv.target.mk:115: Release/obj.target/iconv/binding.o] Error 1
npm ERR! gyp ERR! build error 
npm ERR! gyp ERR! stack Error: `make` failed with exit code: 2
npm ERR! gyp ERR! stack     at ChildProcess.onExit (/home/dgd/nodejs/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:194:23)
npm ERR! gyp ERR! stack     at ChildProcess.emit (node:events:526:28)
npm ERR! gyp ERR! stack     at Process.ChildProcess._handle.onexit (node:internal/child_process:291:12)
npm ERR! gyp ERR! System Linux 5.4.0-110-generic
npm ERR! gyp ERR! command "/home/dgd/nodejs/bin/node" "/home/dgd/nodejs/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
npm ERR! gyp ERR! cwd /home/dgd/nodejs/lib/node_modules/lyrics-searcher/node_modules/iconv
npm ERR! gyp ERR! node -v v16.14.2
npm ERR! gyp ERR! node-gyp -v v8.4.1
npm ERR! gyp ERR! not ok**

**Comment by alias-rahil** [comment]
Try with node v14. Use nvm (node version manager) to maintain and use different versions of node.

**Reply by davegd1387** [reply]
Thanks!
